### PR TITLE
Ds poc - part one

### PR DESCRIPTION
**Pull request by kristenTian**
### Description
Data source POC - p.1
 - Data source view landing page
 - Basic setup
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] init plugin for data source
- [x] public - frontend
  - [x] registry app to stack management
  - [ ] react route and navigation
  - [x ] show data grid/table 
- [ ] server 
   - [ ] api route
   - [ ] CRUD api poc via save object service
      - [ ]  create save object
      - [x] get save object
- [ ] data model for datasource
   - [x]  example of customized saved object

### Note
- As a POC, the UI part is not finalized.
- The constant string reference and comments are not updated yet.

### CODE REVIEW

**Comment by noCharger** [comment]
> did we want a feature branch for this? blocking to avoid accidentally merge.

Let's track PR on - https://github.com/elastic-analytics/OpenSearch-Dashboards

**Review by zhongnansu — CHANGES_REQUESTED** [review]
Let's not submit POC code to the release branch.
Currently MD project POC is supposed to go here https://github.com/elastic-analytics/OpenSearch-Dashboards

**Comment by kristenTian** [comment]
image

**Comment by kristenTian** [comment]
> did we want a feature branch for this? blocking to avoid accidentally merge.

I have closed this pr and moved to the right repo, thanks for your suggestion!

File: src/plugins/data/server/ui_settings.ts [hunk 1]
```diff
@@ -668,6 +668,17 @@ export function getUiSettings(): Record<string, UiSettingsParams<unknown>> {
       }),
       schema: schema.string(),
     },
+    [UI_SETTINGS.DATASOURCE_PLACEHOLDER]: {
```
Comment (zengyan-amazon) [inline src/plugins/data/server/ui_settings.ts:671]: can we put this ui_settings into a data_source management plugin?

Comment (zengyan-amazon) [reply src/plugins/data/server/ui_settings.ts:671]: actually, we may want to have a `data_source` plugin to define the data source saved object type, then another `data_source_management` plugin for the management UI

File: src/plugins/data_source_management/server/saved_objects/sample_data_source.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  mappings: {
+    properties: {
+      name: {
+        type: 'keyword',
+      },
+      count: {
+        type: 'long',
+      },
+    },
+  },
```
Comment (zengyan-amazon) [inline src/plugins/data_source_management/server/saved_objects/sample_data_source.ts:46]: name may need to be `text` to support text based search

what is `count` field? or just for testing purpose

**Pull request closed without merging**